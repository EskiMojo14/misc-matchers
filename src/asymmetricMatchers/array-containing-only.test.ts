describe("arrayContainingOnly", () => {
  it("should expect an array to contain only the expected values", () => {
    expect([1, 2, 2, 3]).toEqual(expect.arrayContainingOnly([1, 2, 3]));

    expect(() => {
      expect([1, 2, 2, 3]).toEqual(expect.arrayContainingOnly([1, 2]));
    }).toThrowErrorMatchingSnapshot();

    expect([1, 2, 2, 3]).toEqual(expect.not.arrayContainingOnly([1, 2]));

    expect(() => {
      expect([1, 2, 2, 3]).toEqual(expect.not.arrayContainingOnly([1, 2, 3]));
    }).toThrowErrorMatchingSnapshot();

    // vs
    expect([1, 2, 2, 3]).toEqual(expect.arrayContaining([1, 2]));
  });
  it("should throw an error if the expected value is not an array", () => {
    expect(() => {
      // @ts-expect-error testing invalid input
      expect([]).toEqual(expect.arrayContainingOnly(1));
    }).toThrowErrorMatchingSnapshot();
  });
  it("should throw an error if the received value is not an array", () => {
    expect(() => {
      expect(1).toEqual(expect.arrayContainingOnly([]));
    }).toThrowErrorMatchingSnapshot();
  });
});
