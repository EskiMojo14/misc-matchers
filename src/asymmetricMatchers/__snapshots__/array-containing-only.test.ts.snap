// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrayContainingOnly should expect an array to contain only the expected values 1`] = `
<d>expect(</b><r>received</g><d>).</b>toEqual<d>(</b><g>expected</g><d>) // deep equality</b>

Expected: <g>arrayContainingOnly<1,2></g>
Received: <r>[1, 2, 2, 3]</g>
`;

exports[`arrayContainingOnly should throw an error if the expected value is not an array 1`] = `Expected value must be an array, not number`;

exports[`arrayContainingOnly should throw an error if the received value is not an array 1`] = `
<d>expect(</b><r>received</g><d>).</b>toEqual<d>(</b><g>expected</g><d>) // deep equality</b>

Expected: <g>arrayContainingOnly<></g>
Received: <r>1</g>
`;
